### PR TITLE
Check action ID uniqueness when creating new ones

### LINKED ISSUE #1502: Make save-action-ids also check them
This will highlight repeated ids and force the user to handle them.

### PR DESCRIPTION
**Pull request by jubeira**
If my understanding is correct, we don't need to check action IDs that we've just generated; we should only check for uniqueness after creation.

Note for the reviewer: hide whitespaces when checking diff.

Closes #1502.

### CODE REVIEW

File: pkg/deployments/hardhat.config.ts [hunk 1]
```diff
@@ -113,63 +113,71 @@ task('check-artifacts', `check that contract artifacts correspond to their build
...
-        await generateActionIdsForTask(taskID);
+        // We're calculating action IDs for whichever contracts we can pull enough information from disk for.
+        // This will calculate action IDs for any contracts which are a named output from a task.
+        for (const taskID of Task.getAllTaskIds()) {
+          await generateActionIdsForTask(taskID);
+        }
       }
+
+      await saveActionIdsTask(args, hre);
+      checkActionIdUniqueness(hre.network.name);
```
Comment (nventuro) [inline pkg/deployments/hardhat.config.ts:180]: Is this because of the early returns?

**Review by nventuro — COMMENTED** [review]
Looks simple enough! Have you tried this to make sure it works?

File: pkg/deployments/hardhat.config.ts (hunk 1 above)
Comment (jubeira) [reply pkg/deployments/hardhat.config.ts:180]: yes, it is.

I considered creating subtasks, but it seemed an overkill. I think this is a better way to put it: first do the original task (whatever the execution path is), then check uniqueness.

**Comment by jubeira** [comment]
I didn't force a collision, but it seems to work:

```
juan@M16:~/prj/bal/pkg/deployments$ hh save-action-ids --id 20221021-managed-pool --network goerli 
  Generating action IDs for ManagedPoolAddRemoveTokenLib of 20221021-managed-pool
  Generating action IDs for CircuitBreakerLib of 20221021-managed-pool
  Generating action IDs for ManagedPoolFactory of 20221021-managed-pool
✅  Verified that no contracts unexpectedly share action IDs
```

**Pull request merged**